### PR TITLE
Picker within a `<form>` submits the first option by default

### PR DESCRIPTION
**Pull request by so99ynoodles**
Closes https://github.com/adobe/react-spectrum/issues/1634

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

- context: https://github.com/adobe/react-spectrum/discussions/1626
- `yarn test packages/@react-spectrum/picker/test/Picker.test.js`

### CODE REVIEW

File: packages/@react-aria/select/src/HiddenSelect.tsx [hunk 1]
```diff
@@ -103,6 +103,7 @@ export function HiddenSelect<T>(props: HiddenSelectProps<T>) {
         <label>
           {label}
           <select {...selectProps}>
+            <option hidden disabled />
```
Comment (LFDanLu) [inline packages/@react-aria/select/src/HiddenSelect.tsx:106]: Whats the reason for the `hidden` and `disabled` here?

**Review by LFDanLu — COMMENTED** [review]
Just one question, but otherwise looks good to me

File: packages/@react-aria/select/src/HiddenSelect.tsx (hunk 1 above)
Comment (so99ynoodles) [reply packages/@react-aria/select/src/HiddenSelect.tsx]: Oh, I forgot to delete them. 
Thank you!

**Review by LFDanLu — APPROVED** [review]
LGTM, thanks for the contribution!

**Review by snowystinger — APPROVED** [review]
Seems reasonable to me, thanks!

**Review by devongovett — APPROVED** [review]

**Pull request merged**